### PR TITLE
[Core] Adding error to function parser

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Part #8776. Adds error to function parser

**Changelog**
- Adds error to function parser

### CODE REVIEW

**Comment by philbucher** [comment]
can you add a test please?

**Comment by loumalouomega** [comment]
> can you add a test please?

Now happy?

**Review by philbucher — DISMISSED** [review]
❤️

**Comment by loumalouomega** [comment]
> heart

Can you fix the test?, I don't understand the error test, I have tried several options, none works

File: kratos/tests/cpp_tests/utilities/test_function_parser_utility.cpp [hunk 1]
```diff
@@ -48,6 +48,9 @@ KRATOS_TEST_CASE_IN_SUITE(GenericFunctionUtility1, KratosCoreFastSuite)
     KRATOS_CHECK_IS_FALSE(function4.UseLocalSystem());
     KRATOS_CHECK_STRING_EQUAL(function4.FunctionBody(), "(cos(x*pi)+sin(y*pi))*t");
     KRATOS_CHECK_DOUBLE_EQUAL(function4.CallFunction(0.25,0.15,0.0,1.5), 1.5*(std::cos(0.25*Globals::Pi) + std::sin(0.15*Globals::Pi)));
+    
+    auto function5 = GenericFunctionUtility("A quien le importa lo que yo haga?A quien le importa lo que yo diga? Yo soy asi, y asi seguire, nunca cambiare");
+    KRATOS_CHECK_EXCEPTION_IS_THROWN(function5.CallFunction(0.0,0.0,0.0,0.0), "Error: Parsing error in function: A quien le importa lo que yo haga?A quien le importa lo que yo diga? Yo soy asi, y asi seguire, nunca cambiare");
```
Comment (philbucher) [inline kratos/tests/cpp_tests/utilities/test_function_parser_utility.cpp]:
```suggestion
    KRATOS_CHECK_EXCEPTION_IS_THROWN(GenericFunctionUtility("A quien le importa lo que yo haga?A quien le importa lo que yo diga? Yo soy asi, y asi seguire, nunca cambiare"), "Error: Parsing error in function: A quien le importa lo que yo haga?A quien le importa lo que yo diga? Yo soy asi, y asi seguire, nunca cambiare");
```

Comment (loumalouomega) [reply kratos/tests/cpp_tests/utilities/test_function_parser_utility.cpp]: But the error is not in construction, it is in the calling..., or maybe it is in construction and i have been wasting time... :P

Comment (philbucher) [reply kratos/tests/cpp_tests/utilities/test_function_parser_utility.cpp]: Parsing must be done only once!
I think this is how it is

Comment (loumalouomega) [reply kratos/tests/cpp_tests/utilities/test_function_parser_utility.cpp]: Indeed, I feel so stupid

**Review by philbucher — APPROVED** [review]

**Pull request merged**